### PR TITLE
doc: transient CRR source

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Replaces https://github.com/scality/backbeat/pull/212.

### CODE REVIEW

File: docs/transient-crr-source.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+configured with CRR, it will stay on the transient source until
+deleted by the user, like if it was a non-transient location.
+
+The replication configuration of buckets which location is transient
+must have a *preferred read location* defined. If not, the replication
+configuration is rejected when set with put-bucket-replication.
+
+### Storage limit
+
+There is a risk to fill up the transient source, in case the backlog
```
Comment (dora-korpar) [inline docs/transient-crr-source.md]: `to fill up` --> `of filling up`

**Review by dora-korpar — APPROVED** [review]

File: docs/transient-crr-source.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+Here are the changes per component:
+
+#### CloudServer
+
+* support for "isTransient" property on locations: enforces that
+  replication configuration of buckets that belong to transient
+  locations specify a *preferred read location*
+
+* new backbeat route `/_/backbeat/batchDelete`, that is used by the
```
Comment (philipyoo) [inline docs/transient-crr-source.md:78]: I guess this should now be `/_/backbeat/api/batchDelete`?

File: docs/transient-crr-source.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  through POST requests containing a JSON body that is an array of
+  locations to delete.
+
+* modified support of *preferred read locations*: now it's set through
+  the replication configuration of a bucket, by appending the
+  `:preferred_read` suffix to the given location in the
+  `<StorageClass>` attribute. This has support from the Orbit user
+  interface through the locations setup.
+
+* added storage limit support through the use of Redis keys storing
```
Comment (philipyoo) [inline docs/transient-crr-source.md:89]: Just curious, are these Redis keys stored permanently in sentinel and removed as needed?

**Review by philipyoo — APPROVED** [review]

File: docs/transient-crr-source.md (hunk 2 above)
Comment (jonathan-gramain) [reply docs/transient-crr-source.md:78]: Actually this is a regular backbeat route that is handled by CloudServer, so does not get proxied to backbeat server.

File: docs/transient-crr-source.md (hunk 3 above)
Comment (jonathan-gramain) [reply docs/transient-crr-source.md:89]: They are never removed right now.

Dora raised an interesting point that if we re-create a location later with the same name, we would re-use the same keys. Maybe this is a strong incentive to cleanup when we delete locations, or alternatively making sure the keys do not conflict when re-creating locations (she should be opening a ticket for this shortly).

**Pull request merged**